sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1111,12 +1111,18 @@ export class DataLakeFileClient extends DataLakePathClient {
   ): Promise<FileUploadResponse> {
     const { span, spanOptions } = createSpan("DataLakeFileClient-upload", options.tracingOptions);
     try {
-      if (isNode && data instanceof Buffer) {
+      if (isNode) {
+        let length = 0;
+        if (data instanceof ArrayBuffer) {
+          length = data.byteLength;
+        } else {
+          length = (data as any).length;
```

Comment:
Why use `any` here? We should avoid using `any` type which will loose benifit of Typing checks.

Reply:
Because the compiler complains
Property 'length' does not exist on type 'Blob' 
Property 'slice' does not exist on type 'ArrayBufferView'